streams/src/main/java/org/apache/kafka/streams/state/internals/ThreadCache.java

```diff
@@ -81,9 +85,11 @@ public synchronized void resize(final long newCacheSizeBytes) {
                 return;
             }
             final CircularIterator<NamedCache> circularIterator = new CircularIterator<>(caches.values());
-            while (sizeBytes() > maxCacheSizeBytes) {
+            while (sizeInBytes.get() > maxCacheSizeBytes) {
                 final NamedCache cache = circularIterator.next();
+                final long oldSize = cache.sizeInBytes();
```

Comment:
probably a relatively unimportant optimization, but we could save two `synchronized` calls by having `#evict` return the number of bytes we evicted from that NamedCache -- just a thought

Reply:
that sounds good to me